Lib/asyncio/selector_events.py

```diff
@@ -735,7 +742,24 @@ def resume_reading(self):
         if self._loop.get_debug():
             logger.debug("%r resumes reading", self)
 
-    def _read_ready(self):
+    def _read_ready_get_buffer(self):
+        if self._conn_lost:
+            return
+
+        try:
+            buf = self._protocol.get_buffer()
```

Comment:
Should this be inside the `try...except`?

Reply:
I think it should. Any unhandled exception in `get_buffer()` means that the protocol is in an inconsistent state, and that it's safer to close the transport and shut everything down.  Speaking of which, we should put `buffer_updated` and `data_received` in similar `try..except` blocks.